### PR TITLE
Make Gazebo simulation works again

### PR DESCRIPTION
**Pull request by olivier-stasse**
The important point is to use the use_sim parameter when loading the URDF.
To make the forward_position_controller working I added some information missing in the yaml file.

Updated the README.md file for the targeted test. 

The other launch files were not tested.

### CODE REVIEW

File: ros2_control_demo_bringup/launch/rrbot_system_position_only_gazebo.launch.py [hunk 1]
```diff
@@ -38,10 +40,18 @@ def generate_launch_description():
     declared_arguments.append(
         DeclareLaunchArgument(
             "use_fake_hardware",
+            default_value="false",
```
Comment (bmagyar) [inline ros2_control_demo_bringup/launch/rrbot_system_position_only_gazebo.launch.py:43]: IMO we should completely remove `use_fake_hardware` from the *gazebo* launch file. We can still explicitly pass `"use_fake_hardware": False,` to the xacro file as there shouldn't be fake hardware present with gazebo...

Comment (olivier-stasse) [reply ros2_control_demo_bringup/launch/rrbot_system_position_only_gazebo.launch.py]: Fixed with fd66e4d

File: ros2_control_demo_bringup/launch/rrbot_system_position_only_gazebo.launch.py [hunk 2]
```diff
@@ -65,21 +67,16 @@ def generate_launch_description():
...
         PythonLaunchDescriptionSource(
             [
-                PathJoinSubstitution(
-                    [
-                        FindPackageShare("gazebo_ros"),
-                        "launch",
-                        "gazebo.launch.py",
-                    ]
-                )
+                os.path.join(get_package_share_directory("gazebo_ros"), "launch"),
```
Comment (destogl) [inline ros2_control_demo_bringup/launch/rrbot_system_position_only_gazebo.launch.py:78]: Why change this? I would not use os.path.jont, but rather `FindPackageShare`. This is basically doing the same as your changes.

**Review by destogl — CHANGES_REQUESTED** [review]
Please delete unused parameter from laucnh and also please don't use "os.joint"

File: ros2_control_demo_bringup/launch/rrbot_system_position_only_gazebo.launch.py [hunk 3]
```diff
@@ -37,11 +38,12 @@ def generate_launch_description():
...
     declared_arguments.append(
         DeclareLaunchArgument(
-            "use_fake_hardware",
+            "use_sim",
             default_value="true",
             description="Start robot with fake hardware mirroring command to its states.",
         )
     )
+
     declared_arguments.append(
```
[the remark this answers is not in the document]
Comment (olivier-stasse) [reply ros2_control_demo_bringup/launch/rrbot_system_position_only_gazebo.launch.py]: Fixed by 0383c6d

File: ros2_control_demo_bringup/launch/rrbot_system_position_only_gazebo.launch.py (hunk 2 above)
Comment (olivier-stasse) [reply ros2_control_demo_bringup/launch/rrbot_system_position_only_gazebo.launch.py]: Fixed by 0383c6d

File: ros2_control_demo_bringup/launch/rrbot_system_position_only_gazebo.launch.py (hunk 1 above)
Comment (bmagyar) [reply ros2_control_demo_bringup/launch/rrbot_system_position_only_gazebo.launch.py]: thanks!

**Review by bmagyar — APPROVED** [review]

**Comment by olivier-stasse** [comment]
Closing in favor of #113 (the new package organization is better)

**Pull request closed without merging**